extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts

```diff
@@ -0,0 +1,383 @@
+			value: constants.server,
+			requiredIndicator: true,
+			width: cssStyles.importDialogLabelWidth
+		}).component();
+
+		const connectionRow = view.modelBuilder.flexContainer().withItems([serverLabel, sourceConnectionTextBox], { flex: '0 0 auto', CSSStyles: { 'margin-right': '10px' } }).withLayout({ flexFlow: 'row', alignItems: 'center' }).component();
+		connectionRow.insertItem(selectConnectionButton, 2, { CSSStyles: { 'margin-right': '0px' } });
+
+		return connectionRow;
+	}
+
+	private createDatabaseRow(view: azdata.ModelView): azdata.FlexContainer {
+		this.sourceDatabaseDropDown = view.modelBuilder.dropDown().withProperties({
+			ariaLabel: constants.databaseNameLabel,
+			required: true,
+			width: cssStyles.importDialogTextboxWidth,
+			editable: true,
+			fireOnTextChange: true
+		}).component();
+
+		this.sourceDatabaseDropDown.onValueChanged(() => {
+			this.setProjectName();
+			this.tryEnableImportButton();
+		});
+
+		const databaseLabel = view.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: constants.databaseNameLabel,
+			requiredIndicator: true,
+			width: cssStyles.importDialogLabelWidth
+		}).component();
+
+		const databaseRow = view.modelBuilder.flexContainer().withItems([databaseLabel, <azdata.DropDownComponent>this.sourceDatabaseDropDown], { flex: '0 0 auto', CSSStyles: { 'margin-right': '10px' } }).withLayout({ flexFlow: 'row', alignItems: 'center' }).component();
+
+		return databaseRow;
+	}
+
+	public setProjectName() {
+		this.projectNameTextBox!.value = newProjectTool.defaultProjectNameFromDb(<string>this.sourceDatabaseDropDown!.value);
+	}
+	private createSourceConnectionComponent(view: azdata.ModelView): azdata.InputBoxComponent {
+		this.sourceConnectionTextBox = view.modelBuilder.inputBox().withProperties({
+			value: '',
+			placeHolder: constants.selectConnection,
+			width: cssStyles.importDialogTextboxWidth,
+			enabled: false
+		}).component();
+
+		this.sourceConnectionTextBox.onTextChanged(() => {
+			this.tryEnableImportButton();
+		});
+
+		return this.sourceConnectionTextBox;
+	}
+
+	private createSelectConnectionButton(view: azdata.ModelView): azdata.Component {
+		this.selectConnectionButton = view.modelBuilder.button().withProperties({
+			ariaLabel: constants.selectConnection,
+			iconPath: IconPathHelper.selectConnection,
+			height: '16px',
+			width: '16px'
+		}).component();
+
+		this.selectConnectionButton.onDidClick(async () => {
+			let connection = await azdata.connection.openConnectionDialog();
+			this.connectionId = connection.connectionId;
+
+			let connectionTextboxValue: string;
+			connectionTextboxValue = this.getConnectionName(connection);
+
+			await this.updateConnectionComponents(connectionTextboxValue, this.connectionId, connection.options.database);
+		});
+
+		return this.selectConnectionButton;
+	}
+
+	private getConnectionName(connection: any): string {
```

Comment:
since the publish dialog also does the same thing, can they use the same function so that only one place needs to be updated in the future if any changes are needed?

Reply:
Done.